flink-yarn/src/main/java/org/apache/flink/yarn/YarnFlinkResourceManager.java

```diff
@@ -78,6 +79,9 @@
 	/** The containers where a TaskManager is starting and we are waiting for it to register */
 	private final Map<ResourceID, YarnContainerInLaunch> containersInLaunch;
 
+	/** The container where a TaskManager has been started and is running in */
+	private final Map<ResourceID, Container> containersLaunched;
```

Comment:
You tell me, since you've authored this component. I guess there was a reason why you clear the list of registered resources when a JM loses its leadership.